pkg/controller/utils.go

```diff
@@ -90,26 +90,31 @@ func SetupSignalHandlerContext() context.Context {
 }
 
 // AddToManagerBuilder aggregates various AddToManager functions.
-type AddToManagerBuilder []func(manager.Manager) error
+type AddToManagerBuilder map[string]func(manager.Manager) error
 
 // NewAddToManagerBuilder creates a new AddToManagerBuilder and registers the given functions.
-func NewAddToManagerBuilder(funcs ...func(manager.Manager) error) AddToManagerBuilder {
-	var builder AddToManagerBuilder
-	builder.Register(funcs...)
+func NewAddToManagerBuilder(params ...interface{}) AddToManagerBuilder {
```

Comment:
Don't modify the `AddToManagerBuilder`: Its responsibility is to accumulate `func(manager.Manager) error` to make it possible to call them in sequence. If you don't want that a function is added, you should just not register it / have several `AddToManager` functions.

Reply:
The point is that I wan't to modify the functions that get called via a command-line flag containing a comma-separated list of things to be disabled. So I turn the builder into a map and pass the keys to be excluded later. Do you see a better way to do this? If yes, could you please explain?

Reply:
I'd suggest the way I described in the comment below - have some dedicated entity that knows what each of these strings means and parse them into a struct and also error if some of these don't exist (see comment with `ControllerSwitchOptions`).

Reply:
The `AddToManagerBuilder` is such a dedicated entity IMO. If I use a different entity, I would have to delete the `AddToManagerBuilder` since it won't be used any more. In this case, I don't see why not using the builder itself?

Reply:
More importantly, I still don't understand what exactly do you suggest. Could you perhaps give an example? We need a mechanism to filter which `AddToManager` functions are added to the manager, and which are not. To avoid repeated ifs for every controller, I'd argue we need a map of names to bools somewhere. Error handling is not hindered, we could just as easily validate on names without having a number of named boolean variables (which will have to be changed every time a new controller is added / removed). Do you have a concrete design in mind? If yes, could you please describe it so that it's understood? If not, may I suggest not to reject a design that is perhaps good enough?

Reply:
I just posted a comment that gives an example that decouples these concerns. The 'map of bools' is there as well (in a slightly changed form).